### PR TITLE
moved validation into static tool

### PR DESCRIPTION
**Pull request by blackbaud-conorwright**
Issue: #8 
Docs: Should probably have documentation on this new class, but I'm not sure where it would go.
The class, `SkyValidationTool`, offers two public static methods: `validateEmailAddress` and `validateUrl`. Each takes in a string value and return a boolean on whether the string is a valid `email address` or `url` respectively.

Would also consider making static methods in the directives themselves

### CODE REVIEW

File: src/app/public/modules/validation-tool/validation-tool.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export class SkyValidationTool {
+  public static validateEmailAddress(emailAddress: string): boolean {
+    // The regex was obtained from http://emailregex.com/
+    // which claims to correctly handle ~99% of all email addresses.
+    // tslint:disable-next-line:max-line-length
+    let regex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return regex.test(emailAddress);
+  }
+
+  public static validateUrl(url: string): boolean {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/validation-tool/validation-tool.ts]: Same as above:
```
public static isUrl(url: string): boolean
```

File: src/app/public/modules/validation-tool/validation-tool.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export class SkyValidationTool {
+  public static validateEmailAddress(emailAddress: string): boolean {
```
[the remark this answers is not in the document]
Comment (blackbaud-conorwright) [reply src/app/public/modules/validation-tool/validation-tool.ts]: I like it! done.

File: src/app/public/modules/validation-tool/validation-tool.ts (hunk 1 above)
Comment (blackbaud-conorwright) [reply src/app/public/modules/validation-tool/validation-tool.ts]: done

File: src/app/public/modules/validation-tool/validation-tool.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export class SkyValidationTool {
```
[the remark this answers is not in the document]
Comment (blackbaud-conorwright) [reply src/app/public/modules/validation-tool/validation-tool.ts]: done. I wasn't that tied to the name `tool` ;)

[the remark this answers is not in the document]
Comment (blackbaud-conorwright) [reply src/app/public/modules/validation-tool/validation-tool.ts]: Good call. Done.

File: src/app/public/modules/email-validation/email-validation.directive.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ import {
   NG_VALIDATORS,
   AbstractControl
 } from '@angular/forms';
+import { SkyValidationTool } from '../validation-tool/validation-tool';
```
[the remark this answers is not in the document]
Comment (blackbaud-conorwright) [reply src/app/public/modules/email-validation/email-validation.directive.ts]: done

File: src/app/public/modules/url-validation/url-validation.directive.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ import {
   NG_VALIDATORS,
   AbstractControl
 } from '@angular/forms';
+import { SkyValidationTool } from '../validation-tool/validation-tool';
```
[the remark this answers is not in the document]
Comment (blackbaud-conorwright) [reply src/app/public/modules/url-validation/url-validation.directive.ts]: done

**Pull request merged**